jni/lib/src/nio/jbyte_buffer.dart

```diff
@@ -0,0 +1,267 @@
+  factory JByteBuffer.allocateDirect(int capacity) {
+    return JByteBuffer.fromRef(
+      Jni.accessors.callStaticMethodWithArgs(
+          _class.reference,
+          _allocateDirectId,
+          JniCallType.objectType,
+          [JValueInt(capacity)]).object,
+    );
+  }
+
+  static final _allocateId = Jni.accessors.getStaticMethodIDOf(
+      _class.reference, r"allocate", r"(I)Ljava/nio/ByteBuffer;");
+
+  /// Allocates a new byte buffer.
+  ///
+  /// Throws:
+  /// * [IllegalArgumentException] - If the capacity is a negative integer
+  factory JByteBuffer.allocate(int capacity) {
+    return const JByteBufferType().fromRef(Jni.accessors
+        .callStaticMethodWithArgs(_class.reference, _allocateId,
+            JniCallType.objectType, [JValueInt(capacity)]).object);
+  }
+
+  static final _wrapWholeId = Jni.accessors.getStaticMethodIDOf(
+      _class.reference, r"wrap", r"([B)Ljava/nio/ByteBuffer;");
+  static final _wrapId = Jni.accessors.getStaticMethodIDOf(
+      _class.reference, r"wrap", r"([BII)Ljava/nio/ByteBuffer;");
+
+  /// Wraps a byte array into a buffer.
+  ///
+  /// The new buffer will be backed by the given byte array; that is,
+  /// modifications to the buffer will cause the array to be modified
+  /// and vice versa.
+  static JByteBuffer wrap(
+    JArray<jbyte> array, [
+    int? offset,
+    int? length,
+  ]) {
+    if (offset == null && length == null) {
+      return const JByteBufferType().fromRef(
+        Jni.accessors.callStaticMethodWithArgs(
+          _class.reference,
+          _wrapWholeId,
+          JniCallType.objectType,
+          [array.reference],
+        ).object,
+      );
+    }
+    offset ??= 0;
+    length ??= array.length - offset;
+    return const JByteBufferType().fromRef(
+      Jni.accessors.callStaticMethodWithArgs(
+        _class.reference,
+        _wrapId,
+        JniCallType.objectType,
+        [array.reference, JValueInt(offset), JValueInt(length)],
+      ).object,
+    );
+  }
+
+  /// Creates a [JByteBuffer] from the content of [list].
+  ///
+  /// The [JByteBuffer] will be allocated using [JByteBuffer.allocateDirect].
+  factory JByteBuffer.fromList(List<int> list) {
+    final buffer = JByteBuffer.allocateDirect(list.length);
+    buffer._asUint8ListUnsafe().setAll(0, list);
+    return buffer;
+  }
+
+  static final _sliceId = Jni.accessors
+      .getMethodIDOf(_class.reference, r"slice", r"()Ljava/nio/ByteBuffer;");
+
+  /// Creates a new byte buffer whose content is a shared subsequence of this
+  /// buffer's content.
+  JByteBuffer slice() {
+    return const JByteBufferType().fromRef(Jni.accessors.callMethodWithArgs(
+        reference, _sliceId, JniCallType.objectType, []).object);
+  }
+
+  static final _duplicateId = Jni.accessors.getMethodIDOf(
+      _class.reference, r"duplicate", r"()Ljava/nio/ByteBuffer;");
+
+  /// Creates a new byte buffer that shares this buffer's content.
```

Comment:
This could benefit from some explanation how this behaves as seen from a Dart programmer perspective.

It calls to Java to do `duplicate`.

Can we add the full documentation from Java instead of only the first line?

https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#duplicate--

I'm not entirely sure how to combine the original docs from Java and the line that from Dart's perspective it's implemented by calling the Java method.

```dart
  /// Creates a new byte buffer that shares this buffer's content.
  ///
  /// Calls the Java method `duplicate` on this object.
  ///
  /// The content of the new buffer will be that of this buffer. Changes to this
  /// buffer's content will be visible in the new buffer; the new buffer itself,
  /// however, will be read-only and will not allow the shared content to be
  /// modified. The two buffers' position, limit, and mark values will be
  /// independent.
  ///
  /// The new buffer's capacity, limit, position, and mark values will be
  /// identical to those of this buffer.
  ///
  /// If this buffer is itself read-only then this method behaves in exactly the
  /// same way as the
  /// [duplicate](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#duplicate--)
  /// method.
  JByteBuffer duplicate() {
    return const JByteBufferType().fromRef(Jni.accessors.callMethodWithArgs(
        reference, _duplicateId, JniCallType.objectType, []).object);
  }
```

or

```dart
  /// Creates a new byte buffer that shares this buffer's content. (Calls the
  /// Java method `duplicate` on this object.)
  ///
  /// The content of the new buffer will be that of this buffer. Changes to this
  /// buffer's content will be visible in the new buffer; the new buffer itself,
  /// however, will be read-only and will not allow the shared content to be
  /// modified. The two buffers' position, limit, and mark values will be
  /// independent.
  ///
  /// The new buffer's capacity, limit, position, and mark values will be
  /// identical to those of this buffer.
  ///
  /// If this buffer is itself read-only then this method behaves in exactly the
  /// same way as the
  /// [duplicate](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#duplicate--)
  /// method.
  JByteBuffer duplicate() {
    return const JByteBufferType().fromRef(Jni.accessors.callMethodWithArgs(
        reference, _duplicateId, JniCallType.objectType, []).object);
  }
```

ditto for other methods from Java.

Reply:
I'll address this in another PR. Other classes like also don't have any documentation. 

https://github.com/dart-lang/jnigen/issues/391

Reply:
Btw the docs you have here is for `asReadOnlyBuffer`.